src/Orchard.Web/Modules/Orchard.Autoroute/Migrations.cs

```diff
@@ -37,5 +52,23 @@ public class Migrations : DataMigrationImpl {
 
             return 3;
         }
+
+        public int UpdateFrom3() {
+
+            SchemaBuilder.AlterTable("AutoroutePartRecord", table => table
+                .AddColumn<bool>("UseCulturePattern", c => c.WithDefault(false))
+            );
+
+            // populate the AutorouteSettings.DefaultPatternDefinitions AutoroutePart settings
+            foreach (var autoroutePart in _contentManager.Query<AutoroutePart, AutoroutePartRecord>().List()) {
+                var settings = autoroutePart.Settings.GetModel<AutorouteSettings>();
+                string patternIndex = autoroutePart.Settings["AutorouteSettings.DefaultPatternIndex"];
+                settings.DefaultPatterns = new List<DefaultPattern> { new DefaultPattern { PatternIndex = patternIndex, Culture = "" } };

```

Comment:
You are right. autoroutePart.Settings["AutorouteSettings.DefaultPatternIndex"]; is still in the database and we can still get it. The migration should take care of it for not seeing any reference of the old setting in the code. If we lazy update it and we program an autogenerateAlias function it will also need to batch update them one by one later on. I prefer having it updated from the start instead of needing to add overhead of validations. But that's your call in the end ;)